src/pybloomfilter.pyx

```diff
@@ -176,6 +178,8 @@ cdef class BloomFilter:
                 else:
                     cpython.PyErr_NoMemory()
 
+        self._writable = self._in_memory or 'w' in mode
```

Comment:
This still doesn't seem right. Mode 'r+' is writable, for example. I think the check should instead be:
`self._in_memory and mode != 'r'`

Reply:
We use the following method to construct the mode:

```python
cdef construct_mode(mode):
    result = os.O_RDONLY
    if 'w' in mode:
        result |= os.O_RDWR
    if 'b' in mode and hasattr(os, 'O_BINARY'):
        result |= os.O_BINARY
    if mode.endswith('+'):
        result |= os.O_CREAT
    return result
```

So, we don't have support for 'a' anyways, and the '+' still doesn't mean we will be able to write to the bloom filter.
If I try this in master, it still produces a segfault:

```python
from pybloomfilter import BloomFilter
BloomFilter(100, 0.1, "temp.bf")
bf1 = BloomFilter.open("temp.bf", mode="r+")
bf1.add("test")
Process finished with exit code 139 (interrupted by signal 11: SIGSEGV)
```

Reply:
This library keeps getting worse each time you look at how it's written. Here, it's checking that the mode _ends_ with '+', instead of just containing it, even though modes like 'r+b' are perfectly valid.

Oh well, I won't insist on this point anyway, since it was just a nitpick.

Reply:
We should rewrite this whole library someday. I think the only reason it makes sense to keep using it is backward-compatibility, but every time we need to change something we really don't know what the author is going to surprise us with.

Reply:
It's a neat library... until you try to read the code.